process/block/preprocess/intermediateResults.go

```diff
@@ -71,7 +72,7 @@ func (irp *intermediateResultsProcessor) CreateAllInterMiniBlocks() []*block.Min
 	}
 
 	for i := 0; i < len(miniBlocks); i++ {
-		if len(miniBlocks[i].TxHashes) > 0 {
+		if miniBlocks[i] != nil && len(miniBlocks[i].TxHashes) > 0 {
```

Comment:
If the answer of the last question is yes, the first test is useless. You can not have nil miniblock on index i.

Reply:
done